onnxruntime/core/session/inference_session.h

```diff
@@ -664,6 +664,17 @@ class InferenceSession {
   std::basic_string<ORTCHAR_T> thread_pool_name_;
   std::basic_string<ORTCHAR_T> inter_thread_pool_name_;
 
+  // This counter increments each time Run() is issued and decrements each time
+  // Run() exits. For concurrent executions this reference counter will be greater than 1
+  // This is currently used by thread-pools to find out if there is a Run() call in progress
+  // so it can adjust it spinning policies.
+  std::atomic_int32_t invocation_refcounter_{0};
```

Comment:
Would the existing current_num_runs_ suffice?

Reply:
I think that one only increments.

Reply:
I will give it a thought.